test/letters/containers/AddressSection.unit.spec.jsx

```diff
@@ -45,6 +46,18 @@ describe('<AddressSection>', () => {
     expect(tree.subTree('.step-content').text()).to.contain('2476 main street, ste #12 west');
   });
 
+  it('should not render an edit button if user not allowed to edit address', () => {
+    const cannotEditProps = { ...defaultProps, canUpdateAddress: false };
+    const component = ReactTestUtils.renderIntoDocument(<AddressSection {...cannotEditProps}/>);
+    expect(() => ReactTestUtils.findRenderedDOMComponentWithTag(component, 'button')).to.throw();
```

Comment:
Does this make sense as an arrow function? Can't we just
```js
expect(ReactTestUtils.findRenderedDOMComponentWithTag(component, 'button')).to.throw();
```
?

Alternatively, what we've done before is:
```js
import { findDOMNode } from 'react-dom';
import { ReactTestUtils } from 'react-dom/test-utils';
...
const component = ReactTestUtils.renderIntoDocument(<AddressSection {...cannotEditProps}/>);
const tree = findDOMNode(component);
expect(tree.querySelectorAll('button.usa-button-outline')).to.be.empty; // or .to.have.lengthOf(0)
```

What you've got is probably a bit faster since we don't have to use `findDOMNode`, but the above I think is clearer that we want to make sure that there are no such buttons. It's also maybe more robust in case `findRenderedDOMComponentWithTag` throws an error for some reason other than not being able to find the element. I have no idea if that's even possible, but paranoid programming. :man_shrugging:

Reply:
As far as I know, the function that's throwing actually has to be wrapped within the `expect()` so that the error's handled / propagated properly. If you run without wrapping, the test itself actually throws an error. If you wrap it like I have but return something other than an error out of the inner function (e.g. `expect(() => true).to.throw()`, the test fails properly, notifying you that it expected to fail but didn't 🎊 

The alternative approach makes sense as well...happy to swap if you think it'd work better!